3 - PB/Documentazione esterna/Specifica Tecnica/Specifica Tecnica.typ

```diff
@@ -255,7 +256,391 @@ Ciò avviene prestando attenzione alle tecnologie impiegate sia nel front-end ch
 
 == Descrizione generale
 
-== Diagramma delle classi
+== Struttura
+Il software WMS3 al fine di perseguire manutenibilità, flessibilità e scalabilità, adotta ed implementa un'architettura "layered", nota anche come "Multi-tier architecture".
+Tale architettura permette di individuare e suddividere la logica del software in 3 principali aspetti, definiti tier (separation of concerns), quali:
+
+- *Persistence layer*: gestisce l'accesso al database e fornisce gli strumenti dedicati alla lettura dei dati al suo interno. I dati letti vengono processati al fine di poter creare gli elementi del Business layer;
+
+- *Business layer*: si occupa di elaborare i dati ricevuti dal layer di persistenza e applicare le regole di business definite. È responsabile di implementare la logica dell'applicazione in modo indipendente dalle tecnologie di persistenza e di presentazione utilizzate;
+
+- *Presentation layer*: permette di trasformare i dati elaborati dal Business layer e le informazioni in una forma comprensibile e accessibile agli utenti finali. Questo include la creazione di interfacce utente grafiche e visualizzazioni 3D degli elementi di interesse.
+
+
+== Design pattern utilizzati
+
+=== Data mapper
+Viene utilizzato per interpretare i dati letti del database mantenendo separate la logica di business dal layer di persistenza. Le classi relative a questo pattern fungono da intermediari tra l'applicazione e la sorgente dati e sono responsabili della conversione delle strutture dati atte alla persistenza in oggetti del dominio dell'applicazione.
```

Comment:
manca la spiegazione di come si comporta, cosa prende in input e cosa produce in output. Prende in input il risultato di una query dal DB in formato JSON e lo traduce nel corrispondente oggetto di business. Quindi spiegherei quali sono i mapper utilizzati già qui o manderei comunque un riferimento alla sezione sotto.

In ogni caso scriverei già qui cosa prendono in input e cosa producono in output (so che è parzialmente già detto, ma preferirei più dettagliato)

Reply:
non elencherei qua le classi relative al pattern se no si dovrebbe fare anche con gli altri pattern e si andrebbe a descrivere la maggior parte delle classi qua al posto che nella sezione classi e componenti